### PR TITLE
[docs] add docstring in torch.cuda.get_device_properties

### LINKED ISSUE #49737: torch.cuda.get_device_properties is missing in cuda docs
## 📚 Documentation

`torch.cuda.get_device_properties` is missing in both https://pytorch.org/docs/stable/cuda.html and https://pytorch.org/docs/master/cuda.html while we can still use the API.

Since it has no docstring, it doesn't show in the docs. Docs source:
```
torch.cuda
===================================

.. currentmodule:: torch.cuda

.. automodule:: torch.cuda
   :members:
```

cc @jlin27 @mruberry

### PR DESCRIPTION
**Pull request by jeffydc**
Fixes #49737 

Added docstring in `torch.cuda.get_device_properties`
Added the `Returns` in `torch.cuda.get_device_name`

### CODE REVIEW

**Comment by ydcjeff** [comment]
I added the docstring similar to other `get_device_*`.
Let me know if it needs to be improved.

**Comment by ydcjeff** [comment]
Just noticed the return types are missing after the function/class name (annotated with →) on the master docs.
That is present on stable docs.

File: torch/cuda/__init__.py [hunk 1]
```diff
@@ -293,6 +296,15 @@ def get_device_capability(device: Optional[_device_t] = None) -> Tuple[int, int]
 
 
 def get_device_properties(device: _device_t) -> _CudaDeviceProperties:
+    r"""Gets the properties of a device.
+
+    Arguments:
```
Comment (ngimel) [inline torch/cuda/__init__.py:301]: Arguments: -> Args:

**Review by ngimel — APPROVED** [review]
Looks good, please change `Arguments` to `Args`.

File: torch/cuda/__init__.py (hunk 1 above)
Comment (ydcjeff) [inline torch/cuda/__init__.py:301]:
```suggestion
    Args:
```

**Comment by ydcjeff** [comment]
any updates to merge?

**Pull request closed without merging**